ehb_service/apps/api/views/relation.py

```diff
@@ -1,26 +1,30 @@
 import json
 
-from restlib2.resources import Resource
+from rest_framework.views import APIView
+from rest_framework.response import Response
+from rest_framework import status
+from rest_framework.decorators import permission_classes
+from rest_framework import permissions
 
 from core.models.identities import Relation, PedigreeSubjectRelation
 from core.forms import PedigreeSubjectRelationForm
 from api.helpers import FormHelpers
 
 
-class RelationResource(Resource):
-    supported_accept_types = ['application/json']
-    model = 'core.models.identities.Relation'
+@permission_classes((permissions.AllowAny,))
+class RelationView(APIView):
 
     def get(self, request, **kwargs):
         relations = Relation.objects.all()
         d = []
         for relation in relations:
             d.append(relation.to_dict())
 
-        return (json.dumps(d))
+        return Response(d)
 
 
-class PedigreeSubjectRelationResource(Resource):
+@permission_classes((permissions.AllowAny,))
```

Comment:
Instead of 'AllowAny' should this be 'IsAuthenticated'?

The AllowAny permission class will allow unrestricted access, regardless of if the request was authenticated or unauthenticated.

This permission is not strictly required, since you can achieve the same result by using an empty list or tuple for the permissions setting, but you may find it useful to specify this class because it makes the intention explicit.

Reply:
Removed the @permissionsAllowAny header from all the classes. Created a api/authentication.py file to override Django authentication so that we can continue authenticating using the api tokens that we've created instead of switching over to Django-generated auth-tokens. 803fea7